### PR TITLE
Enable Checkstyle fail on error / violation for Key Vault

### PR DESCRIPTION
**Pull request by JonathanGiles**

### CODE REVIEW

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -26,11 +26,24 @@ jobs:
         JavaVersion: '1.11'
 
   steps:
+  - task: Maven@3
+    # This build task can be removed once we start publishing the build tools to Maven Central
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply .azure-pipelines/client.yml]: If that is possible it would be better....

[the remark this answers is not in the document]
Comment (weshaggard) [reply .azure-pipelines/client.yml]: @JonathanGiles I thought we were planning to let the Analyze step handle running checkstyle? There isn't any reason to run it in both "Build" and "Analyze" jobs as a failure in either will cause the overall build to fail.

[the remark this answers is not in the document]
Comment (JonathanGiles) [reply .azure-pipelines/client.yml]: My concern was that checkstyle run as part of maven site might not generate a build failure. I don't have bandwidth to double check this right now though - @conniey or @g2vinay - if you have any bandwidth are you able to confirm that running the `mvn site:site site:stage` with fail on error set to true does indeed fail the build? If not, I will return to this ASAP.

[the remark this answers is not in the document]
Comment (g2vinay) [reply .azure-pipelines/client.yml]: I ran mvn site:site site:stage locally by setting failsOnError and failsOnViolation to true for checkstyle plugin.
It caused the build failure locally.

[the remark this answers is not in the document]
Comment (JonathanGiles) [reply .azure-pipelines/client.yml]: In this case I will tweak the PR to remove the extra checkstyle run. Thanks for checking.

**Pull request merged**